src/firewheel/control/model_component_manager.py

```diff
@@ -681,12 +685,14 @@ def _print_plugin_initialization_help(
             f"\n[cyan]{filled_sig}[/cyan]"
         )
 
-    def build_experiment_graph(self):
+    def build_experiment_graph(self, dry_run=False):
         """
         Builds the experiment graph by processing all the model components
 
         Returns:
             list: A list of errors that were reported when trying to execute.
+            dry_run (bool): Indicates whether the model components should be run (:py:data:`False`)
+                or simply imported (i.e. checked for syntax errors). Defaults to :py:data:`False`.
 
```

Comment:
I guess linters are useful sometimes...

```suggestion

        Args:
            dry_run (bool): Indicates whether the model components should be run (:py:data:`False`)
                or simply imported (i.e. checked for syntax errors). Defaults to :py:data:`False`.
                
        Returns:
            list: A list of errors that were reported when trying to execute.


```